shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm

```diff
@@ -182,6 +183,81 @@
   return canvases;
 }
 
+void FlutterPlatformViewsController::CompositeWithParams(
+    int view_id,
+    const flutter::EmbeddedViewParams& params) {
+  UIView* touch_interceptor = touch_interceptors_[view_id].get();
+
+  CGRect frame = CGRectMake(0, 0, params.sizePoints.width(), params.sizePoints.height());
+  touch_interceptor.frame = frame;
+  UIView* lastView = touch_interceptor;
+
+  // Reset everything related to transform.
+  lastView.layer.transform = CATransform3DIdentity;
+  ResetAnchor(lastView.layer);
+
+  // Start loop to apply transforms/clips.
+  std::vector<EmbeddedViewMutator>::reverse_iterator iter = params.transformStack->rbegin();
+  int64_t clipCount = 0;
+  while (iter != params.transformStack->rend()) {
+    switch (iter->type()) {
+      case transform: {
+        CATransform3D transform = GetCATransform3DFromSkMatrix(iter->matrix());
+        lastView.layer.transform = CATransform3DConcat(lastView.layer.transform, transform);
+        break;
+      }
+      case clip_rect:
+      case clip_rrect:
+      case clip_path: {
+        UIView* view = lastView.superview;
+        // if we need more clips operations than last time, create a new view.
+        ++clipCount;
+        if (clipCount > clip_count_[view_id]) {
+          [lastView removeFromSuperview];
+          view = [[UIView alloc] initWithFrame:lastView.bounds];
+          [view addSubview:lastView];
+        }
+        PerformClip(view, iter->type(), iter->rect(), iter->rrect(), iter->path());
+        ResetAnchor(view.layer);
+        lastView = view;
+        break;
+      }
+    }
+    ++iter;
+  }
+
+  // Reverse scale based on screen scale.
+  //
+  // The UIKit frame is set based on the logical resolution instead of physical.
+  // (https://developer.apple.com/library/archive/documentation/DeviceInformation/Reference/iOSDeviceCompatibility/Displays/Displays.html).
+  // However, flow is based on the physical resolution. For eaxmple, 1000 pixels in flow equals
+  // 500 points in UIKit. And until this point, we did all the calculation based on the flow
+  // resolution. So we need to scale down to match UIKit's logical resolution.
+  CGFloat screenScale = [UIScreen mainScreen].scale;
+  lastView.layer.transform = CATransform3DConcat(
+      lastView.layer.transform, CATransform3DMakeScale(1 / screenScale, 1 / screenScale, 1));
+
+  // If we have less cilp operations this time, remove unnecessary views.
+  // We skip this process if we have more clip operations this time.
+  int64_t extraClipsFromLastTime = clip_count_[view_id] - clipCount;
+  if (extraClipsFromLastTime > 0 && lastView.superview) {
+    [lastView removeFromSuperview];
+    UIView* superview = lastView.superview;
+    while (extraClipsFromLastTime > 0) {
+      UIView* superSuperView = superview.superview;
+      [superview removeFromSuperview];
+      [superview release];
+      superview = superSuperView;
+    }
+    [flutter_view_.get() addSubview:lastView];
```

Comment:
I think this might change the composition order in an unwanted way when we have multiple platform views, e.g [P_1, P_2] (in this order), and now the clip chain for P_1 changes, we remove the chain for P_1 and add it back to flutter_view_ which results in flipping the order([P_2, P]).

Reply:
Done